Source/ACE.Server/WorldObjects/Player_Inventory.cs

```diff
@@ -941,7 +941,7 @@ public void HandleActionGiveObjectRequest(ObjectGuid targetID, ObjectGuid itemGu
         {
             
             WorldObject target = CurrentLandblock.GetObject(targetID) as WorldObject;
-            WorldObject item = GetInventoryItem(itemGuid) as WorldObject;
+            WorldObject item = GetInventoryItem(itemGuid) as WorldObject ?? GetWieldedItem(itemGuid) as WorldObject; 
```

Comment:
Is it certain that GetWieldedItem() covers all of the possible scenarios?

Perhaps item should be checked for null after this line, and return if no item was found?

Reply:
I would agree that to be safe a null check should be added.